### ISSUE TITLE
NPE in Query.equals() when Query isn't attached

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12502]
The current default implementation of `AbstractQueryPart.equals()` produces a NPE when the `Query` is not attached, e.g.

```java
var s1 = selectOne();
var s2 = selectOne();
s1.detach();
s1.equals(s2);
```

Produces:

```
java.lang.NullPointerException: Cannot invoke "org.jooq.Configuration.dsl()" because the return value of "org.jooq.impl.AbstractQueryPart.configuration()" is null
	at org.jooq.impl.AbstractQueryPart.equals(AbstractQueryPart.java:170)
```

This hasn't appeared a lot, because `selectOne()` doesn't produce a detached query, it's attached to the internal default configuration.

**Comment by lukaseder** [comment]
This fix causes a javac compiler bug (?) to manifest in a totally unrelated place:
https://twitter.com/lukaseder/status/1447824955522129923

But only on Open JDK 11.0.11 on Linux, not on Windows, and only on the 3.14 branch, not 3.16 and 3.15 (which are built with a more recent version of the JDK), and only the trial edition, not the others (some of which are built with Java 8 on branch 3.14). Testing Open JDK 11.0.12 for now. Otherwise, I'll just revert the fix for 3.14

**Comment by lukaseder** [comment]
Luckily, it seems that the bug cannot be reproduced with 11.0.12
